### PR TITLE
Palpatim/update pods

### PR DESCRIPTION
**Pull request by palpatim**
*Description of changes:*
- Update AWS dependencies to 2.13.1
- Add SwiftLint as a Pod dependency
- Update CHANGELOG for #379

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWSAppSyncClient.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2049,7 +2049,6 @@
 				B48056BC22A992E200E4F742 /* AppSyncSubscriptionWithSyncTests.swift in Sources */,
 				FAC28D3B221B0F09000F84CD /* CacheKeyTests.swift in Sources */,
 				2171809123FDB28100E520C9 /* UserPoolsBasedConnectionPoolTests.swift in Sources */,
-				B47DECF6235C027F003A8127 /* APIKeyAuthInterceptorTests.swift in Sources */,
```
Comment (royjit) [inline AWSAppSyncClient.xcodeproj/project.pbxproj:2052]: Why was this file removed?

Comment (palpatim) [reply AWSAppSyncClient.xcodeproj/project.pbxproj:2052]: It wasn't removed with this commit; it must have been Xcode removing stray file references.
